src/sqlitetablemodel.cpp

```diff
@@ -600,41 +605,48 @@ void SqliteTableModel::buildQuery()
 
         for(QMap<int, QString>::const_iterator i=m_mWhere.constBegin();i!=m_mWhere.constEnd();++i)
         {
-            QString column;
-            if(m_vDisplayFormat.size())
-                column = QString("col%1").arg(i.key());
-            else
-                column = m_headers.at(i.key());
-            where.append(QString("%1 %2 AND ").arg(sqlb::escapeIdentifier(column)).arg(i.value()));
+            QString column = sqlb::escapeIdentifier(m_headers.at(i.key()));
+            where.append(QString("%1 %2 AND ").arg(column).arg(i.value()));
```

Comment:
This won't work as expected. The filters will then be applied to the original data, not the converted data. E.g. using the Julian day format leads to this SQL:
```sql
SELECT datetime(`my_date_column`) AS `my_date_column` WHERE `my_date_column` LIKE '%2017%';
```
vs this with the old code:
```sql
SELECT datetime(`my_date_column`) AS `col1` WHERE `col1` LIKE '%2017%';
```

In SQLite the results will be different because in the first case the WHERE part is applied first and the datetime function will only be applied on the resulting data set afterwards.

Reply:
Ok, there's no easy way :cry: I can fall back to my first idea that was not saving the display formats, because I don't know how can it be made to work preserving the original column names. I don't like either adding a prefix like `new_` or just `_` to all the columns.